### PR TITLE
Adjust content type condition to evaluate to true when no content is displayed

### LINKED ISSUE #15452: Content Type layer rule behaves unexpectedly on pages where no content is displayed with "Detail" display type
### Describe the bug
If a page does not contain any content items rendered with the "Detail" display type, all Content Type layer rules will evaluate to false. I would expect some Content Type rules to evaluate to true in this case, for example "Content type does not equal "MadeUpContentType"".

### To Reproduce
1. Cook a fresh site with "Agency" recipe
2. Enable Search and Lucene features.
3. Add a search index
image
4. Add a new layer with a single Content Type rule configured with "Does not equal" as the operator and "MadeUpContentType" as the value.
image

5. Add a search widget to the Footer zone and configure it to use the search index and layer added previously.
image

6. On the front end of the site, search for anything
7. Notice that the search widget does not appear in the footer

### Expected behavior
I would expect the search widget to still be rendered in this case because there are no content items of type "MadeUpContentType" rendered on the search results page.

In ContentTypeConditionEvaluatorDriver.EvaluateAsync, it seems like operatorComparer.Compare should be called with a blank value in the event that the _contentTypes hash set is empty. This would cause operators like "Does not equal", "Does not start with", "Does not end with", and "Does not contain" to return true, which seems reasonable, whild "Equals", "Starts with", "Ends with", and "Contains" will still evaluate to false, as they do currently.

### PR DESCRIPTION
**Pull request by rwawr**
Adjust content type condition to evaluate to true when no content is displayed and the operation for the given condition is a negation to address. Fixes #15452.

### CODE REVIEW

**Comment by Piedone** [comment]
This is a follow-up from https://github.com/OrchardCMS/OrchardCore/pull/15602. And now there's no policy bot... Anyway, you accepted the CLA.

**Review by Piedone — APPROVED** [review]
This seems quite clear-cut to me, but I'm not sure if I'm missing something obvious. @deanmarcussen could you please also take a quick look?

File: src/OrchardCore.Modules/OrchardCore.Rules/Drivers/ContentTypeConditionEvaluatorDriver.cs [hunk 1]
```diff
@@ -43,6 +44,14 @@ public ValueTask<bool> EvaluateAsync(Condition condition)
         private ValueTask<bool> EvaluateAsync(ContentTypeCondition condition)
         {
             var operatorComparer = _operatorResolver.GetOperatorComparer(condition.Operation);
+
+            // If no content types are considered, use the empty string as the value to compare against,
+            // since we still want comparisons such as "Does Not Equal", "Does Not Start With", etc. to evaluate to true in this case.
+            if (!_contentTypes.Any())
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Rules/Drivers/ContentTypeConditionEvaluatorDriver.cs:50]:
```suggestion
            if (_contentTypes.Count == 0)
```

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Rules/Drivers/ContentTypeConditionEvaluatorDriver.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 using System.Collections.Generic;
+using System.Linq;
```
Comment (rwawr) [inline src/OrchardCore.Modules/OrchardCore.Rules/Drivers/ContentTypeConditionEvaluatorDriver.cs:2]:
```suggestion
```

**Pull request merged**